src/Core/InstanceConfigTrait.php

```diff
@@ -68,11 +68,12 @@
      *
      * @param string|array|null $key The key to get/set, or a complete array of configs.
      * @param mixed|null $value The value to set.
-     * @param bool $merge Whether to merge or overwrite existing config, defaults to true.
+     * @param bool|string $merge 'deep' to merge recursively, 'shallow' for simple merge,
+     *   false to overwrite, defaults to 'deep'.
      * @return mixed Config value being read, or the object itself on write operations.
      * @throws \Cake\Core\Exception\Exception When trying to set a key that is invalid.
      */
-    public function config($key = null, $value = null, $merge = true)
+    public function config($key = null, $value = null, $merge = 'deep')
```

Comment:
@ndm2 Yeah I found out the same workaround [here](http://stackoverflow.com/questions/24357985/php-traits-defining-generic-constants#answer-24377563). Just not too sure if we should do that..